### DISCUSSION TITLE
read offline syslog log files with SO

### DISCUSSION THREAD

**Discussion by vijaytejak** [Security-Onion-Solutions/securityonion#6756] [Unsupported Versions]
HIi,

I have syslogs being saved in a file, i want to generate analytics with kibana, the log format is supported it is ntopng syslogs, how to do that.

is there is solution like, copying the log file to security onion and running a command to view the analytics in kibana

**Comment by weslambert** [comment]
Assuming each log is on it's own line, you could try using the Filebeat's file input to read them and point them to the syslog pipeline, or something like the following (streaming to port 514):

```
while read line; 
  do echo -n "$line" | nc -v -w 1 localhost 514; 
done < $file
```

If you have multiple files, maybe something like (untested, back of napkin code):

```
for file in $file1 $file2 $file3;
do
  while read line; 
    do echo -n "$line" | nc -v -w 1 localhost 514; 
  done < $file
done
```
Keep in mind, the ingestion timestamps might be different than the event timestamps.
